schema.md

```diff
@@ -158,11 +158,64 @@ same database, the duplicate entry could be written using only the `id`,
 The `related` field gives a list of IDs of closely related vulnerabilities, such
 as the same problem in alternate ecosystems. 
 
-### package field
+### summary, details fields
+
+The `summary` field gives a one-line, English textual summary of the
+vulnerability. It is recommended that this field be kept short, on the order of
+no more than 120 characters.
+
+The `details` field gives additional English textual details about the
+vulnerability. The field is plain text. Newline characters must be considered
+line breaks when displaying the text, but the display need not use a fixed-width
+font, and the text should not assume one.
+
+The `summary` field is plain text.
+
+The `details` field is CommonMark markdown (a subset of GitHub-Flavored
+Markdown). Display code may at its discretion sanitize the input further, such
+as stripping raw HTML and links that do not start with http:// or https://.
+Databases are encouraged not to include those in the first place. (The goal is
+to balance flexibility of presentation with not exposing vulnerability database
+display sites to unnecessary vulnerabilities.)
+
+### affected fields
+
+The `affected` field is a JSON array containing objects that describes the
+affected packages versions, meaning those that contain the vulnerability.
+
+Within each object in the `affected` array, the `package` field identifies the
+package containing the vulnerability. In most cases, there should be exactly one
+entry in the `affected` array per affected `package` to describe all affected
+versions. In rare cases, for example if the `ecosystem_specific` encodes
+platform information that doesn't apply equally to all listed versions and
+ranges, a separate entry with the same `package` in the `affected` array may be
+used.
+
+The `versions` field can enumerate a specific set of affected versions, and the
```

Comment:
I wonder if we should collapse ranges and versions down to

   "versions": [{"type":..., "version":..., "before":...}] 

like in https://github.com/CVEProject/cve-schema/issues/87#issuecomment-893978107, with `"type": "exact"` for a single-version range, both to match that and to avoid having the two different fields that have to be merged by implementations.

On the other hand then you'd have to say that it's OK to have "exact" versions overlap non-semver ranges, to make them available to tooling.

Reply:
Would prefer to keep the current explicit separation for our schema. Enforcing the that "versions" has to always exist if not SEMVER is much easier this way (as you mention below).

It should be pretty trivial for tooling to convert between the two, and too much on top of handling the other differences.